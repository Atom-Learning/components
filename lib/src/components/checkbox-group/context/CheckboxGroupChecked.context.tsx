import * as React from 'react'

import type { CheckboxGroupItemValue } from '../CheckboxGroup.types'

type CheckboxGroupCheckedContextValue = {
  checked: CheckboxGroupItemValue[]
  handleItemCheckedChange: (
    newItemChecked: boolean,
    itemValue: CheckboxGroupItemValue
  ) => void
}

type CheckboxGroupCheckedContextProps = {
  checked?: CheckboxGroupCheckedContextValue['checked']
  defaultChecked?: CheckboxGroupCheckedContextValue['checked']
  onCheckedChange?: (
    checked: CheckboxGroupCheckedContextValue['checked']
  ) => void
}

export const CheckboxGroupCheckedContext =
  React.createContext<CheckboxGroupCheckedContextValue>({
    checked: [],
    handleItemCheckedChange: () => null
  })

const generateNewCheckedFn = (
  currentChecked: CheckboxGroupCheckedContextValue['checked']
) => {
  return (newItemChecked, itemValue) => {
    const newCheckedSet = new Set(currentChecked)
    newItemChecked
      ? newCheckedSet.add(itemValue)
      : newCheckedSet.delete(itemValue)
    const newChecked = Array.from(newCheckedSet)
    return newChecked
  }
}

export const CheckboxGroupCheckedProvider = ({
  checked: controlledChecked,
  defaultChecked = [],
  onCheckedChange,
  ...rest
}: CheckboxGroupCheckedContextProps) => {
  const [checked, setChecked] = React.useState(defaultChecked)

  const handleItemControlledCheckedChange = React.useCallback(
    (newItemChecked, itemValue) => {
      if (!controlledChecked) return () => null
      const newCheckedFn = generateNewCheckedFn(controlledChecked)
      const newChecked = newCheckedFn(newItemChecked, itemValue)
      onCheckedChange?.(newChecked)
    },
    [onCheckedChange, controlledChecked]
  )

  const handleItemCheckedChange = React.useCallback(
    (newItemChecked, itemValue) => {
      setChecked((prevChecked) => {
        const newCheckedFn = generateNewCheckedFn(prevChecked)
        const newChecked = newCheckedFn(newItemChecked, itemValue)
        onCheckedChange?.(newChecked)
        return newChecked
      })
    },
    [onCheckedChange]
  )

  const value = React.useMemo<CheckboxGroupCheckedContextValue>(() => {
    const isControlled = Array.isArray(controlledChecked)
    return {
      checked: isControlled
        ? (controlledChecked as CheckboxGroupCheckedContextValue['checked']) // We're literally JUST checking it above but TS still flags it, wrongly, so casting
        : checked,
      handleItemCheckedChange: isControlled
        ? handleItemControlledCheckedChange
        : handleItemCheckedChange
    }
  }, [
    checked,
    handleItemCheckedChange,
    handleItemControlledCheckedChange,
    controlledChecked
  ])

  return <CheckboxGroupCheckedContext.Provider value={value} {...rest} />
}
