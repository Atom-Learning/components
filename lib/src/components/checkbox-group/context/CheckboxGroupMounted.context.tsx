import * as React from 'react'

import type { CheckboxGroupItemValue } from '../CheckboxGroup.types'

type CheckboxGroupMountedContextValue = {
  mounted: CheckboxGroupItemValue[]
  handleItemMountedChange: (
    newItemMounted: boolean,
    itemValue: CheckboxGroupItemValue
  ) => void
}

export const CheckboxGroupMountedContext =
  React.createContext<CheckboxGroupMountedContextValue>({
    mounted: [],
    handleItemMountedChange: () => null
  })

export const CheckboxGroupMountedProvider = (props) => {
  const { handleItemMountedChange: handleItemMountedChangeFurtherUpContext } =
    React.useContext(CheckboxGroupMountedContext)

  const [mounted, setMounted] = React.useState<CheckboxGroupItemValue[]>([])
  const handleItemMountedChange = React.useCallback(
    (newItemMounted, itemValue) => {
      handleItemMountedChangeFurtherUpContext?.(newItemMounted, itemValue)
      setMounted((prevMounted) => {
        const newMountedSet = new Set(prevMounted)
        newItemMounted
          ? newMountedSet.add(itemValue)
          : newMountedSet.delete(itemValue)
        const newMounted = Array.from(newMountedSet)
        return newMounted
      })
    },
    [handleItemMountedChangeFurtherUpContext]
  )

  const value = React.useMemo<CheckboxGroupMountedContextValue>(
    () => ({ mounted, handleItemMountedChange }),
    [mounted, handleItemMountedChange]
  )

  return <CheckboxGroupMountedContext.Provider value={value} {...props} />
}
