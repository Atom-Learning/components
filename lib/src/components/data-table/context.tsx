import * as React from 'react'
import {
  useReactTable,
  getCoreRowModel,
  getPaginationRowModel,
  getSortedRowModel,
  Table
} from '@tanstack/react-table'

type DataTableContextType<T = unknown> = Table<T> & {
  applyPagination: () => void
  getTotalRows: () => number
}
const DataTableContext =
  React.createContext<DataTableContextType<unknown> | null>(null)

type TableProviderProps = {
  columns
  data: Array<Record<string, unknown>>
  children: React.ReactNode
}

export const DataTableProvider = ({
  columns,
  data,
  children
}: TableProviderProps): JSX.Element => {
  const [isPaginated, setIsPaginated] = React.useState<boolean>(false)

  const applyPagination = React.useCallback(() => {
    setIsPaginated(true)
  }, [])

  const getTotalRows = () => data.length

  const table = useReactTable<unknown>({
    columns,
    data,
    getCoreRowModel: getCoreRowModel(),
    getPaginationRowModel: isPaginated ? getPaginationRowModel() : undefined,
    getSortedRowModel: getSortedRowModel()
  })

  const value = React.useMemo(() => {
    return {
      ...table,
      applyPagination,
      getTotalRows
    }
  }, [table, applyPagination, getTotalRows])

  return (
    <DataTableContext.Provider value={value}>
      {children}
    </DataTableContext.Provider>
  )
}

export const useDataTable = <T extends Record<string, unknown>>() => {
  const context = React.useContext(DataTableContext) as DataTableContextType<T>

  if (!context)
    throw new Error(
      'useDataTable can only be called from inside a DataTableProvider'
    )

  return context
}
