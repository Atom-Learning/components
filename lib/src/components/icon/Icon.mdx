---
title: Icon
component: Icon
description: A standardised way to render icons, ensuring that only the icons you use are included in your output
category: Content
---

`Icon` is a light wrapper component that provides a standardised set of styles to the icon in use. It requires an SVG from our icon library to be passed into the `is` prop.

```tsx preview
<Stack gap="4">
  <Icon is={BatteryMedium} />
  <Icon is={Crossing} />
  <Icon is={EyeCrossed} />
  <Icon is={Paperclip} />
  <Icon is={Sun} />
  <Icon is={Wheelchair} />
</Stack>
```

Ensure that you pair the imported icon (in this case `Brightness`) with the `Icon` component to render in a consistent and reliable way. Review the available icons on the [icon listing page](/theme/icons)

```tsx
import { Icon } from '@components'
import { Brightness } from '@atom-learning/icons'

const Component = () => (
  <Icon is={Brightness} size="lg" css={{ color: '$primary' }} />
)
```
