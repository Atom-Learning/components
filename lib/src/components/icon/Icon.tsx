import * as React from 'react'

import { styled } from '~/stitches'
import { Override } from '~/utilities/types'

export const StyledIcon = styled('svg', {
  display: 'inline-block',
  fill: 'none',
  flexShrink: 0,
  stroke: 'currentcolor',
  strokeLinecap: 'round',
  strokeLinejoin: 'round',
  verticalAlign: 'middle',
  variants: {
    size: {
      sm: { size: '$1', strokeWidth: '1.5' },
      md: { size: '$2', strokeWidth: '1.75' },
      lg: { size: '$3', strokeWidth: '2' }
    }
  }
})

type IconProps = Override<
  React.ComponentProps<typeof StyledIcon>,
  {
    is: React.FC<React.SVGProps<SVGSVGElement>>
    as?: never
  }
>

export const Icon = React.forwardRef<SVGSVGElement, IconProps>(
  ({ is: SVG, size = 'md', ...remainingProps }, ref) => (
    <StyledIcon
      size={size}
      aria-hidden="true"
      {...remainingProps}
      as={SVG}
      ref={ref}
    />
  )
)
