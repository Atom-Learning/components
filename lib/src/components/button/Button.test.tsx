import { Ok } from '@atom-learning/icons'
import { fireEvent, render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

import { Icon } from '../icon'
import { Button } from '.'
import { ComponentsProvider } from '../../context'

describe(`Button component`, () => {
  const props = { onClick: jest.fn() }

  it('renders a button', async () => {
    const { container } = render(<Button {...props}>BUTTON</Button>)

    await screen.getByText('BUTTON')

    expect(container).toMatchSnapshot()
  })

  it('renders a button - has no programmatically detectable a11y issues', async () => {
    const { container } = render(<Button {...props}>BUTTON</Button>)

    expect(await axe(container)).toHaveNoViolations()
  })

  it('renders a outline button', async () => {
    const { container } = render(
      <Button appearance="outline" {...props}>
        BUTTON
      </Button>
    )

    await screen.getByText('BUTTON')
    expect(container).toMatchSnapshot()
  })

  it('renders a outline button - has no programmatically detectable a11y issues', async () => {
    const { container } = render(
      <Button appearance="outline" {...props}>
        BUTTON
      </Button>
    )

    expect(await axe(container)).toHaveNoViolations()
  })

  it('renders a disabled button', async () => {
    const { container } = render(
      <Button disabled {...props}>
        BUTTON
      </Button>
    )

    const button = await screen.getByText('BUTTON')

    expect(button).toBeDisabled()
    expect(container).toMatchSnapshot()
  })

  it('renders a disabled button - has no programmatically detectable a11y issues', async () => {
    const { container } = render(
      <Button disabled {...props}>
        BUTTON
      </Button>
    )

    expect(await axe(container)).toHaveNoViolations()
  })

  it('renders a disabled warning outline button', async () => {
    const { container } = render(
      <Button disabled appearance="outline" theme="warning" {...props}>
        BUTTON
      </Button>
    )

    const button = await screen.getByText('BUTTON')

    expect(button).toBeDisabled()
    expect(container).toMatchSnapshot()
  })

  it('renders a outline button with secondary theme', async () => {
    const { container } = render(
      <Button appearance="outline" theme="secondary" {...props}>
        BUTTON
      </Button>
    )

    expect(container).toMatchSnapshot()
  })

  it('renders a disabled warning outline button - has no programmatically detectable a11y issues', async () => {
    const { container } = render(
      <Button disabled appearance="outline" theme="warning" {...props}>
        BUTTON
      </Button>
    )

    expect(await axe(container)).toHaveNoViolations()
  })

  it('renders a button with an icon', async () => {
    const { container } = render(
      <Button {...props}>
        BUTTON <Icon is={Ok} />
      </Button>
    )

    await screen.getByText('BUTTON')

    expect(container).toMatchSnapshot()
  })

  it('renders a button with an icon - has no programmatically detectable a11y issues', async () => {
    const { container } = render(
      <Button {...props}>
        BUTTON <Icon is={Ok} />
      </Button>
    )

    await screen.getByText('BUTTON')

    expect(await axe(container)).toHaveNoViolations()
  })

  it('onClick is called when button is clicked', async () => {
    const handleClick = jest.fn()
    render(<Button onClick={handleClick}>BUTTON</Button>)

    fireEvent.click(screen.getByRole('button'))

    expect(handleClick).toHaveBeenCalled()
  })

  it('is polymorphic', async () => {
    render(
      <Button as="a" href="https://app.atomlearning.co.uk">
        BUTTON
      </Button>
    )

    const link = screen.getByRole('link')

    expect(link).toBeInTheDocument()
    expect(link).toHaveAttribute('href', 'https://app.atomlearning.co.uk')
    expect(link).toHaveAttribute('target', '_blank')
    expect(link).toHaveAttribute('rel', 'noopener noreferrer')
    expect(link).not.toHaveAttribute('type', 'button')
  })

  it('renders an anchor if provided a link', async () => {
    render(<Button href="/somewhere">ATOM</Button>)

    const link = screen.getByRole('link')

    expect(link).toHaveAttribute('href', '/somewhere')
    expect(link).not.toHaveAttribute('target')
    expect(link).not.toHaveAttribute('rel')
  })

  describe('Loading state', () => {
    it('renders a loading button', async () => {
      const { container } = render(
        <Button isLoading {...props}>
          BUTTON
        </Button>
      )

      expect(await screen.getByRole('alert')).toBeInTheDocument()

      expect(container).toMatchSnapshot()
      expect(await axe(container)).toHaveNoViolations()
    })

    it('renders a loading button - has no programmatically detectable a11y issues', async () => {
      const { container } = render(
        <Button isLoading {...props}>
          BUTTON
        </Button>
      )

      expect(await axe(container)).toHaveNoViolations()
    })

    it('onClick is prevented if button is in loading state', async () => {
      const handleClick = jest.fn()
      render(
        <Button onClick={handleClick} isLoading>
          BUTTON
        </Button>
      )

      fireEvent.click(screen.getByRole('button'))

      expect(handleClick).toHaveBeenCalledTimes(0)
    })

    it('works with ComponentsProvider', async () => {
      const { container } = render(
        <ComponentsProvider Link={(props) => <p {...props} />}>
          <Button href="https://google.com/">EXTERNAL</Button>
          <Button href="/hello">INTERNAL</Button>
        </ComponentsProvider>
      )

      expect(container.querySelector('p')).toBeInTheDocument()
      expect(screen.getByRole('link')).toBeInTheDocument()
    })
  })
})
