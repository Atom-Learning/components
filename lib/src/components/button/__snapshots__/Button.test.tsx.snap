// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Loading state renders a loading button 1`] = `
@media  {
  .c-kaaUSm {
    align-items: center;
    background: unset;
    border: unset;
    border-radius: var(--radii-0);
    cursor: pointer;
    display: flex;
    font-family: var(--fonts-body);
    font-weight: 600;
    justify-content: center;
    padding: unset;
    text-decoration: none;
    transition: all 100ms ease-out;
    white-space: nowrap;
    width: max-content;
  }

  .c-kaaUSm[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-dbrbZt {
    display: inline-block;
    fill: none;
    stroke: currentcolor;
    stroke-linecap: round;
    stroke-linejoin: round;
    vertical-align: middle;
  }

  .c-dhzjXW {
    display: flex;
  }

  .c-jBOHFZ {
    animation-name: k-gJNjLs;
    animation-duration: 1s;
    animation-fill-mode: both;
    animation-iteration-count: infinite;
    animation-timing-function: linear;
    background-color: currentColor;
    border-radius: var(--radii-round);
  }

  .c-jBOHFZ:nth-child(1) {
    animation-delay: -300ms;
  }

  .c-jBOHFZ:nth-child(2) {
    animation-delay: -150ms;
  }

  .c-jBOHFZ:nth-child(3) {
    animation-delay: 0;
  }

  .c-gTMqnE {
    align-items: center;
    display: flex;
    justify-content: center;
    visibility: hidden;
  }
}

@media  {
  .c-kaaUSm-dXsdwv-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
    height: var(--sizes-4);
    padding-left: var(--space-5);
    padding-right: var(--space-5);
    gap: var(--space-3);
  }

  .c-kaaUSm-dXsdwv-size-md .c-dbrbZt {
    height: 20px;
    width: 20px;
  }

  .c-dbrbZt-dMrjnF-size-md {
    height: var(--sizes-2);
    width: var(--sizes-2);
    stroke-width: 1.75;
  }

  .c-kaaUSm-DdIlq-isLoading-true {
    cursor: not-allowed;
    opacity: 0.6;
    pointer-events: none;
  }

  .c-dhzjXW-bICGYT-justify-center {
    justify-content: center;
  }

  .c-jBOHFZ-jIYkAt-size-md {
    height: 6px;
    width: 6px;
    margin-left: 2px;
    margin-right: 2px;
  }

  .c-gTMqnE-fVlWzK-size-md {
    gap: var(--space-3);
  }
}

@media  {
  .c-kaaUSm-dfkxSu-cv {
    background: var(--colors-primary);
    color: white;
  }

  .c-kaaUSm-dfkxSu-cv[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-kaaUSm-dfkxSu-cv:not([disabled]):hover,
  .c-kaaUSm-dfkxSu-cv:not([disabled]):focus {
    background: var(--colors-primaryMid);
    color: white;
  }

  .c-kaaUSm-dfkxSu-cv:not([disabled]):active {
    background: var(--colors-primaryDark);
  }

  .c-kaaUSm-fAhdck-cv {
    border: 1px solid;
    border-color: currentColor;
    color: var(--colors-primary);
  }

  .c-kaaUSm-fAhdck-cv[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-kaaUSm-fAhdck-cv:not([disabled]):hover,
  .c-kaaUSm-fAhdck-cv:not([disabled]):focus {
    text-decoration: none;
    color: var(--colors-primaryMid);
  }

  .c-kaaUSm-fAhdck-cv:not([disabled]):active {
    color: var(--colors-primaryDark);
  }

  .c-kaaUSm-ctZbUF-cv {
    border: 1px solid;
    border-color: currentColor;
    color: var(--colors-primaryDark);
  }

  .c-kaaUSm-ctZbUF-cv[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-kaaUSm-ctZbUF-cv:not([disabled]):hover,
  .c-kaaUSm-ctZbUF-cv:not([disabled]):focus {
    text-decoration: none;
    color: hsla(228, 82%, 25%, 1);
  }

  .c-kaaUSm-ctZbUF-cv:not([disabled]):active {
    color: hsla(228, 82%, 20%, 1);
  }
}

@media  {
  .c-dhzjXW-ibMCfDd-css {
    position: absolute;
  }
}

<div>
  <button
    class="c-kaaUSm c-kaaUSm-dXsdwv-size-md c-kaaUSm-DdIlq-isLoading-true c-kaaUSm-dfkxSu-cv"
    type="button"
  >
    <div
      class="c-dhzjXW c-dhzjXW-bICGYT-justify-center c-dhzjXW-ibMCfDd-css"
      role="alert"
    >
      <span
        style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
      >
        Loading
      </span>
      <div
        class="c-jBOHFZ c-jBOHFZ-jIYkAt-size-md"
      />
      <div
        class="c-jBOHFZ c-jBOHFZ-jIYkAt-size-md"
      />
      <div
        class="c-jBOHFZ c-jBOHFZ-jIYkAt-size-md"
      />
    </div>
    <span
      class="c-gTMqnE c-gTMqnE-fVlWzK-size-md"
    >
      BUTTON
    </span>
  </button>
</div>
`;

exports[`Button component renders a button 1`] = `
@media  {
  .c-kaaUSm {
    align-items: center;
    background: unset;
    border: unset;
    border-radius: var(--radii-0);
    cursor: pointer;
    display: flex;
    font-family: var(--fonts-body);
    font-weight: 600;
    justify-content: center;
    padding: unset;
    text-decoration: none;
    transition: all 100ms ease-out;
    white-space: nowrap;
    width: max-content;
  }

  .c-kaaUSm[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }
}

@media  {
  .c-kaaUSm-dXsdwv-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
    height: var(--sizes-4);
    padding-left: var(--space-5);
    padding-right: var(--space-5);
    gap: var(--space-3);
  }

  .c-kaaUSm-dXsdwv-size-md .c-dbrbZt {
    height: 20px;
    width: 20px;
  }
}

@media  {
  .c-kaaUSm-dfkxSu-cv {
    background: var(--colors-primary);
    color: white;
  }

  .c-kaaUSm-dfkxSu-cv[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-kaaUSm-dfkxSu-cv:not([disabled]):hover,
  .c-kaaUSm-dfkxSu-cv:not([disabled]):focus {
    background: var(--colors-primaryMid);
    color: white;
  }

  .c-kaaUSm-dfkxSu-cv:not([disabled]):active {
    background: var(--colors-primaryDark);
  }
}

<div>
  <button
    class="c-kaaUSm c-kaaUSm-dXsdwv-size-md c-kaaUSm-dfkxSu-cv"
    type="button"
  >
    BUTTON
  </button>
</div>
`;

exports[`Button component renders a button with an icon 1`] = `
@media  {
  .c-kaaUSm {
    align-items: center;
    background: unset;
    border: unset;
    border-radius: var(--radii-0);
    cursor: pointer;
    display: flex;
    font-family: var(--fonts-body);
    font-weight: 600;
    justify-content: center;
    padding: unset;
    text-decoration: none;
    transition: all 100ms ease-out;
    white-space: nowrap;
    width: max-content;
  }

  .c-kaaUSm[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-dbrbZt {
    display: inline-block;
    fill: none;
    stroke: currentcolor;
    stroke-linecap: round;
    stroke-linejoin: round;
    vertical-align: middle;
  }
}

@media  {
  .c-kaaUSm-dXsdwv-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
    height: var(--sizes-4);
    padding-left: var(--space-5);
    padding-right: var(--space-5);
    gap: var(--space-3);
  }

  .c-kaaUSm-dXsdwv-size-md .c-dbrbZt {
    height: 20px;
    width: 20px;
  }

  .c-dbrbZt-dMrjnF-size-md {
    height: var(--sizes-2);
    width: var(--sizes-2);
    stroke-width: 1.75;
  }
}

@media  {
  .c-kaaUSm-dfkxSu-cv {
    background: var(--colors-primary);
    color: white;
  }

  .c-kaaUSm-dfkxSu-cv[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-kaaUSm-dfkxSu-cv:not([disabled]):hover,
  .c-kaaUSm-dfkxSu-cv:not([disabled]):focus {
    background: var(--colors-primaryMid);
    color: white;
  }

  .c-kaaUSm-dfkxSu-cv:not([disabled]):active {
    background: var(--colors-primaryDark);
  }

  .c-kaaUSm-fAhdck-cv {
    border: 1px solid;
    border-color: currentColor;
    color: var(--colors-primary);
  }

  .c-kaaUSm-fAhdck-cv[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-kaaUSm-fAhdck-cv:not([disabled]):hover,
  .c-kaaUSm-fAhdck-cv:not([disabled]):focus {
    text-decoration: none;
    color: var(--colors-primaryMid);
  }

  .c-kaaUSm-fAhdck-cv:not([disabled]):active {
    color: var(--colors-primaryDark);
  }

  .c-kaaUSm-ctZbUF-cv {
    border: 1px solid;
    border-color: currentColor;
    color: var(--colors-primaryDark);
  }

  .c-kaaUSm-ctZbUF-cv[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-kaaUSm-ctZbUF-cv:not([disabled]):hover,
  .c-kaaUSm-ctZbUF-cv:not([disabled]):focus {
    text-decoration: none;
    color: hsla(228, 82%, 25%, 1);
  }

  .c-kaaUSm-ctZbUF-cv:not([disabled]):active {
    color: hsla(228, 82%, 20%, 1);
  }
}

<div>
  <button
    class="c-kaaUSm c-kaaUSm-dXsdwv-size-md c-kaaUSm-dfkxSu-cv"
    type="button"
  >
    BUTTON 
    <svg
      aria-hidden="true"
      class="c-dbrbZt c-dbrbZt-dMrjnF-size-md"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="4 13 9 18 20 7"
      />
    </svg>
  </button>
</div>
`;

exports[`Button component renders a disabled button 1`] = `
@media  {
  .c-kaaUSm {
    align-items: center;
    background: unset;
    border: unset;
    border-radius: var(--radii-0);
    cursor: pointer;
    display: flex;
    font-family: var(--fonts-body);
    font-weight: 600;
    justify-content: center;
    padding: unset;
    text-decoration: none;
    transition: all 100ms ease-out;
    white-space: nowrap;
    width: max-content;
  }

  .c-kaaUSm[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }
}

@media  {
  .c-kaaUSm-dXsdwv-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
    height: var(--sizes-4);
    padding-left: var(--space-5);
    padding-right: var(--space-5);
    gap: var(--space-3);
  }

  .c-kaaUSm-dXsdwv-size-md .c-dbrbZt {
    height: 20px;
    width: 20px;
  }
}

@media  {
  .c-kaaUSm-dfkxSu-cv {
    background: var(--colors-primary);
    color: white;
  }

  .c-kaaUSm-dfkxSu-cv[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-kaaUSm-dfkxSu-cv:not([disabled]):hover,
  .c-kaaUSm-dfkxSu-cv:not([disabled]):focus {
    background: var(--colors-primaryMid);
    color: white;
  }

  .c-kaaUSm-dfkxSu-cv:not([disabled]):active {
    background: var(--colors-primaryDark);
  }

  .c-kaaUSm-fAhdck-cv {
    border: 1px solid;
    border-color: currentColor;
    color: var(--colors-primary);
  }

  .c-kaaUSm-fAhdck-cv[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-kaaUSm-fAhdck-cv:not([disabled]):hover,
  .c-kaaUSm-fAhdck-cv:not([disabled]):focus {
    text-decoration: none;
    color: var(--colors-primaryMid);
  }

  .c-kaaUSm-fAhdck-cv:not([disabled]):active {
    color: var(--colors-primaryDark);
  }
}

<div>
  <button
    class="c-kaaUSm c-kaaUSm-dXsdwv-size-md c-kaaUSm-dfkxSu-cv"
    disabled=""
    type="button"
  >
    BUTTON
  </button>
</div>
`;

exports[`Button component renders a disabled warning outline button 1`] = `
@media  {
  .c-kaaUSm {
    align-items: center;
    background: unset;
    border: unset;
    border-radius: var(--radii-0);
    cursor: pointer;
    display: flex;
    font-family: var(--fonts-body);
    font-weight: 600;
    justify-content: center;
    padding: unset;
    text-decoration: none;
    transition: all 100ms ease-out;
    white-space: nowrap;
    width: max-content;
  }

  .c-kaaUSm[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }
}

@media  {
  .c-kaaUSm-dXsdwv-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
    height: var(--sizes-4);
    padding-left: var(--space-5);
    padding-right: var(--space-5);
    gap: var(--space-3);
  }

  .c-kaaUSm-dXsdwv-size-md .c-dbrbZt {
    height: 20px;
    width: 20px;
  }
}

@media  {
  .c-kaaUSm-dfkxSu-cv {
    background: var(--colors-primary);
    color: white;
  }

  .c-kaaUSm-dfkxSu-cv[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-kaaUSm-dfkxSu-cv:not([disabled]):hover,
  .c-kaaUSm-dfkxSu-cv:not([disabled]):focus {
    background: var(--colors-primaryMid);
    color: white;
  }

  .c-kaaUSm-dfkxSu-cv:not([disabled]):active {
    background: var(--colors-primaryDark);
  }

  .c-kaaUSm-fAhdck-cv {
    border: 1px solid;
    border-color: currentColor;
    color: var(--colors-primary);
  }

  .c-kaaUSm-fAhdck-cv[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-kaaUSm-fAhdck-cv:not([disabled]):hover,
  .c-kaaUSm-fAhdck-cv:not([disabled]):focus {
    text-decoration: none;
    color: var(--colors-primaryMid);
  }

  .c-kaaUSm-fAhdck-cv:not([disabled]):active {
    color: var(--colors-primaryDark);
  }
}

<div>
  <button
    class="c-kaaUSm c-kaaUSm-dXsdwv-size-md"
    disabled=""
    type="button"
  >
    BUTTON
  </button>
</div>
`;

exports[`Button component renders a outline button 1`] = `
@media  {
  .c-kaaUSm {
    align-items: center;
    background: unset;
    border: unset;
    border-radius: var(--radii-0);
    cursor: pointer;
    display: flex;
    font-family: var(--fonts-body);
    font-weight: 600;
    justify-content: center;
    padding: unset;
    text-decoration: none;
    transition: all 100ms ease-out;
    white-space: nowrap;
    width: max-content;
  }

  .c-kaaUSm[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }
}

@media  {
  .c-kaaUSm-dXsdwv-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
    height: var(--sizes-4);
    padding-left: var(--space-5);
    padding-right: var(--space-5);
    gap: var(--space-3);
  }

  .c-kaaUSm-dXsdwv-size-md .c-dbrbZt {
    height: 20px;
    width: 20px;
  }
}

@media  {
  .c-kaaUSm-dfkxSu-cv {
    background: var(--colors-primary);
    color: white;
  }

  .c-kaaUSm-dfkxSu-cv[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-kaaUSm-dfkxSu-cv:not([disabled]):hover,
  .c-kaaUSm-dfkxSu-cv:not([disabled]):focus {
    background: var(--colors-primaryMid);
    color: white;
  }

  .c-kaaUSm-dfkxSu-cv:not([disabled]):active {
    background: var(--colors-primaryDark);
  }

  .c-kaaUSm-fAhdck-cv {
    border: 1px solid;
    border-color: currentColor;
    color: var(--colors-primary);
  }

  .c-kaaUSm-fAhdck-cv[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-kaaUSm-fAhdck-cv:not([disabled]):hover,
  .c-kaaUSm-fAhdck-cv:not([disabled]):focus {
    text-decoration: none;
    color: var(--colors-primaryMid);
  }

  .c-kaaUSm-fAhdck-cv:not([disabled]):active {
    color: var(--colors-primaryDark);
  }
}

<div>
  <button
    class="c-kaaUSm c-kaaUSm-dXsdwv-size-md c-kaaUSm-fAhdck-cv"
    type="button"
  >
    BUTTON
  </button>
</div>
`;

exports[`Button component renders a outline button with secondary theme 1`] = `
@media  {
  .c-kaaUSm {
    align-items: center;
    background: unset;
    border: unset;
    border-radius: var(--radii-0);
    cursor: pointer;
    display: flex;
    font-family: var(--fonts-body);
    font-weight: 600;
    justify-content: center;
    padding: unset;
    text-decoration: none;
    transition: all 100ms ease-out;
    white-space: nowrap;
    width: max-content;
  }

  .c-kaaUSm[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }
}

@media  {
  .c-kaaUSm-dXsdwv-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
    height: var(--sizes-4);
    padding-left: var(--space-5);
    padding-right: var(--space-5);
    gap: var(--space-3);
  }

  .c-kaaUSm-dXsdwv-size-md .c-dbrbZt {
    height: 20px;
    width: 20px;
  }
}

@media  {
  .c-kaaUSm-dfkxSu-cv {
    background: var(--colors-primary);
    color: white;
  }

  .c-kaaUSm-dfkxSu-cv[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-kaaUSm-dfkxSu-cv:not([disabled]):hover,
  .c-kaaUSm-dfkxSu-cv:not([disabled]):focus {
    background: var(--colors-primaryMid);
    color: white;
  }

  .c-kaaUSm-dfkxSu-cv:not([disabled]):active {
    background: var(--colors-primaryDark);
  }

  .c-kaaUSm-fAhdck-cv {
    border: 1px solid;
    border-color: currentColor;
    color: var(--colors-primary);
  }

  .c-kaaUSm-fAhdck-cv[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-kaaUSm-fAhdck-cv:not([disabled]):hover,
  .c-kaaUSm-fAhdck-cv:not([disabled]):focus {
    text-decoration: none;
    color: var(--colors-primaryMid);
  }

  .c-kaaUSm-fAhdck-cv:not([disabled]):active {
    color: var(--colors-primaryDark);
  }

  .c-kaaUSm-ctZbUF-cv {
    border: 1px solid;
    border-color: currentColor;
    color: var(--colors-primaryDark);
  }

  .c-kaaUSm-ctZbUF-cv[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-kaaUSm-ctZbUF-cv:not([disabled]):hover,
  .c-kaaUSm-ctZbUF-cv:not([disabled]):focus {
    text-decoration: none;
    color: hsla(228, 82%, 25%, 1);
  }

  .c-kaaUSm-ctZbUF-cv:not([disabled]):active {
    color: hsla(228, 82%, 20%, 1);
  }
}

<div>
  <button
    class="c-kaaUSm c-kaaUSm-dXsdwv-size-md c-kaaUSm-ctZbUF-cv"
    type="button"
  >
    BUTTON
  </button>
</div>
`;
